python/tempo/tsdf.py

```diff
@@ -157,38 +155,39 @@ def __addPrefixToColumns(self, col_list: list[str], prefix: str) -> "TSDF":
         """
         Add prefix to all specified columns.
         """
-        if prefix != "":
-            prefix = prefix + "_"
+        # no-op if no prefix
+        if not prefix:
+            return self
 
-        df = reduce(
-            lambda df, idx: df.withColumnRenamed(
-                col_list[idx], "".join([prefix, col_list[idx]])
-            ),
-            range(len(col_list)),
-            self.df,
+        # build a column rename map
+        col_map = {col: "_".join([prefix, col]) for col in col_list}
+        # df = self.df.withColumnsRenamed(col_map)
+
+        # build a list of column expressions to rename columns in a select
+        rename_fn = (
+            lambda col: sfn.col(col).alias(col_map[col])
+            if col in col_map
+            else sfn.col(col)
         )
+        select_exprs = [rename_fn(col) for col in self.df.columns]
```

Comment:
```suggestion
        select_exprs = [
            sfn.col(col).alias(col_map[col] if col in col_map
            else sfn.col(col)
            for col in self.df.columns
        ]
```

Refactor to get rid of lambda and b/c this doesn't seem to be used a lot in other places in the code so probably not worth a function